Наставник. Разметка. CSS

MargoBayus opened the pull request:
CSS естественно не полностью :)

MargoBayus committed:
Наставник. Разметка. CSS

CSS естественно не полностью :)

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
     <meta charset="utf-8">
     <title>HTML Academy: Девайс</title>
+    <link rel="icon" href="favicon.ico">
+    <link rel="stylesheet" href="styles/style.css">
```
Давай добавим мета-теги странице (https://sc.link/kV5v)

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
   <body>
+    <header class="page-header">
+      
+      <nav class="navigation-logo">
```
Мне кажется что не нужно оборачивать логотип в тег `<nav>`, т.к. это не совсем про навигацию по сайту

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+        <a href="index.html"><img src="Images/logo-header.svg" alt="Логотип Device"></a>
+        </nav>
+        <nav class="navigation-list">
+          <form>
```
Для формы нужно добавить методы action и method

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+        </nav>
+        <nav class="navigation-list">
+          <form>
+            <input class="search" type="search" name="q" placeholder="Текст поиска">
```
Лучше давать осмысленные 'name' для инпутов, так тебе самой будет проще разбирать свой код через неделю/две

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+        <nav class="navigation-list">
+          <form>
+            <input class="search" type="search" name="q" placeholder="Текст поиска">
+            <input class="search-submit" type="submit" value="Найти">
```
Это должно быть кнопкой а не полем

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+
+            <div class="basket">  
+              <a class="navigation-link" href="#">Корзина</a>
+            </div> 
```
Мне кажется что обёртки `<div>` для этих ссылок лишние, т.к. с их помощью ты ничего не группируешь. Классы с обёрток можно перевесить на сами ссылки

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+            </div> 
+            
+
+          <ul class="main-menu">
```
Это не совсем один список. "Каталог товаров" - это отдельный элемент который лучше вынести из списка

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+</p>
+  <a class="button-main" href="#">Подробнее</a>
+
+  <div class="description-card-01">
```
Здесь лучше использовать теги для списка терминов - https://up.htmlacademy.ru/htmlcss/34/book/1.14

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+      <div class="category-item">
+
+    <div id="square"> <img src="Images/icon-1.svg" class="item-1">
+      <H3><a href="#">Виртуальная реальность</a> </H3>  
```
Теги пишутся с маленькой буквы. Если использовать форматтер кода, то он будет самостоятельно исправлять такие ошибки

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+
+      <div class="category-item">
+
+    <div id="square"> <img src="Images/icon-1.svg" class="item-1">
```
Не понимаю зачем здесь id. Объясни пожалуйста

MargoBayus replied:
id здесь для того, чтобы вписать в квадрат иконку. или можно было просто сделать класс и прописать те же характеристики?

AlSudar replied:
Да, для этого классы и нужны. Id используется как правило для создания якорей и взаимодействия с JS

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+
+
+
+<section class="DGK">
```
Классы пишутся с маленькой буквы и без аббревиатур

AlSudar commented on index.html:
```diff
@@ -3,10 +3,254 @@
+  <p>Мы с удовольствием доставим ваш товар прямо
+  к вашему подъезду совершенно бесплатно! Ведь
+  мы неплохо заработаем, поднимая его на ваш этаж.</p>
+<img src="Images/delivery.svg" width="136px" height="164px" alt="Доставка">
```
Мне кажется ты опечаталсь, и случайно написала с большой буквы

AlSudar commented on styles/style.css:
```diff
@@ -0,0 +1,157 @@
+
+
+/*800 raleway*/
+@font-face{
```
Шрифты обычно подключают наверху всего кода, чтобы когда браузер дочитал до объявления шрифта, он уже загрузил их

AlSudar commented on styles/style.css:
```diff
@@ -0,0 +1,157 @@
+  line-height: 50px;
+}
+
+h2{
```
Стили лучше добавлять не тегу а классу, если у тебя на странице будет несколько разных по стилям тегов h2, возникнет путаница с стилями

AlSudar commented:
Привет! Спасибо за дз, ты проделала большую работу, почти полностью создала разметку всей страницы, и стили для неё. Я оставил несколько замечаний по коду, пожалуйста прочти и подумай над ними. Также пожалуйста при работе пользуйся форматтером кода. Успехов в учёбе!

MargoBayus committed:
Корректировки по комментариям

Там ещё немного внесла дополнительного материала, что не успела сделать до проверки.
По поводу почему Images везде было с большой буквы - это не опечатка, проблема была в том, что при создании папки, я по привычке написала её с большой буквы. Так что это тоже исправила)

MargoBayus committed:
1-2 неделя

Я уже если честно запуталась относительно недель, но по сути у меня не сделана анимация кнопок, поп-ап, и есть вопросы по слайдеру. мне кажется я сделала там что-то не так, поэтому нужна помощь в общем и целом. заранее благодарствую)))

AlSudar commented on index.html:
```diff
@@ -141,50 +155,59 @@ <H3><a href="#">Квадрокоптеры</a></H3>
 </div>
+<div class="vertical-rectangle"></div>
 <div class="delivery-description">
   <H1>Доставка</H1>
```
Теги пишутся с маленькой буквы

AlSudar commented on index.html:
```diff
@@ -141,50 +155,59 @@ <H3><a href="#">Квадрокоптеры</a></H3>
+  
+<section class="about">
+  <div class="rectangle"></div>
+   <H1>О нас</H1>
```
Теги пишутся с маленькой буквы

AlSudar commented on index.html:
```diff
@@ -141,50 +155,59 @@ <H3><a href="#">Квадрокоптеры</a></H3>
-
+<section class="contacts">
+  <div class="rectangle"></div>
   <H1>Контакты</H1>
```
Теги пишутся с маленькой буквы

AlSudar commented on index.html:
```diff
@@ -141,50 +155,59 @@ <H3><a href="#">Квадрокоптеры</a></H3>
     Заодно сможете проверить работоспособность покупки. </p>
 
-    <H5>Интернет-магазин «Девайс»</H5>
+    <H3>Интернет-магазин «Девайс»</H3>
```
Теги пишутся с маленькой буквы

AlSudar commented on index.html:
```diff
@@ -141,50 +155,59 @@ <H3><a href="#">Квадрокоптеры</a></H3>
 
-    <H5>Интернет-магазин «Девайс»</H5>
+    <H3>Интернет-магазин «Девайс»</H3>
     <p>Санкт-Петербург, набережная реки Карповки, 5, литера П.</p>
```
Тут лучше использовать тег adress

AlSudar commented on index.html:
```diff
@@ -194,63 +217,63 @@ <h5>Время работы</h5>
 
-    <form>
+<div class="form-describe">
+   <form>
```
Формам нужно добавлять атрибуты action и method, без них они (формы) не будут работать (подробнее - https://up.htmlacademy.ru/htmlcss/34/module/9)

AlSudar commented on index.html:
```diff
@@ -194,63 +217,63 @@ <h5>Время работы</h5>
+   <form>
       <input class="email" type="email" placeholder="Ваш e-mail">
-      <input class="email-submit" type="submit" value="Подписаться">
+      <button class="email-button" type="button"> Подписаться</button>
```
Здесь лучше кнопке задать type submit, иначе форма не отправится

AlSudar commented on styles/style.css:
```diff
@@ -3,89 +3,183 @@
 
+.promo-gallery{
+  display:inline-flex;
+  margin-top: -180px;
```
Лучше не использовать отрицательные отступы, т.к. в таких случаях остальные элементы ведут себя непредсказуемо

AlSudar commented on styles/style.css:
```diff
@@ -3,89 +3,183 @@
 
-  font-size: 36px;
+.promo-gallery-description{
+  margin-top: -470px;
```
Лучше не использовать отрицательные отступы, т.к. в таких случаях остальные элементы ведут себя непредсказуемо

AlSudar reviewed (commented):
Привет! Пожалуйста, перед тем как переходить к следующему дз, обработай всей комментарии по предыдущему, чтобы я мог закрыть пуллреквест (и принять тем самым дз) и не создавалась путаница. Сейчас страница уже начинает приобретать необходимые для финального результата очертания, но вижу что нет анимаций кнопок, поп-апа и не сделан слайдер. Ты можешь задать вопросы по тем местам которые не понимаешь в общем чате с однокурсниками, или в одном из наших общих созвонов

p.s. пожалуйста, включи редактирование кода. Я показывал во время созвона как это сделать (https://disk.yandex.ru/i/pZ9tJJ22a3BK-A). Займёт не больше двух минут
